### ISSUE TITLE
Mac user 

### ISSUE THREAD

**Issue by HatmanAlchemy** [Nebukam/PCGExtendedToolkit#36]
Bonjour Timothé. I would like to use this on my Mac but Xcode gives a "declaration shadows a field of " error on line 86 of PCGExDataFiler error. Unfortunately I am more a visual artist than coder. I wondered if this is an easy fix ( it runs fine on my Windows machine ) Xcode is very fussy. Thanks
![Screenshot 2024-03-31 at 11 07 51 am](https://github.com/Nebukam/PCGExtendedToolkit/assets/21061155/8e42ed33-de40-43b0-a00f-7da3fa3050fb)

**Comment by Nebukam** [comment]
Hey there @HatmanAlchemy ! Thanks for logging the issue; unfortunately I'm on vacation at the moment and don't have access to a computer where I could fix this -- it should definitely be an easy fix tho. 

You'll have to wait for sometime after April 8th tho 😅

**Comment by HatmanAlchemy** [comment]
No problem. Enjoy your vacation. Merci.

**Comment by Nebukam** [comment]
@HatmanAlchemy pushed a fix! It's possible that more error pop up after this fix as I have currently no way to test on macOS; this bug may hide more mac-specific issues. They're usually trivial to fix, so please let met know if you encounter any more errors; otherwise feel free to mark the issue as closed <3

Thanks again for taking the time to report the problem! :D

**Comment by HatmanAlchemy** [comment]
Still get the same error on the same line 86 
![Screenshot 2024-04-09 at 4 12 06 PM](https://github.com/Nebukam/PCGExtendedToolkit/assets/21061155/7ebe8385-5374-436a-9ece-b201b859fae3)

**Comment by Nebukam** [comment]
Yeah I jumped on the gun and fixed a non-issue. The problem was unrelated to the commit -- coming back from vacations heh :D

**Comment by Nebukam** [comment]
@HatmanAlchemy any news on this?

**Comment by HatmanAlchemy** [comment]
Nope just the same. 
![Screenshot 2024-04-16 at 7 17 47 AM](https://github.com/Nebukam/PCGExtendedToolkit/assets/21061155/fd9c1d0b-e850-4049-b520-fafb77c4e496)

**Comment by HatmanAlchemy** [comment]
I get this now
![Screenshot 2024-04-16 at 8 53 41 AM](https://github.com/Nebukam/PCGExtendedToolkit/assets/21061155/9c62a6cd-07f2-4f15-8333-2be06e996f63)

**Comment by HatmanAlchemy** [comment]
Showing Recent Messages
ExternalBuildToolExecution PCG_EX_C_Build (in target 'PCG_EX_C_Build' from project 'PCG_EX_C (Mac)')
    cd /Users/Shared/Epic\ Games/UE_5.3
    export ACTION\=
    export ALLOW_TARGET_PLATFORM_SPECIALIZATION\=NO
    export ALTERNATE_GROUP\=staff
    export ALTERNATE_MODE\=u+w,go-w,a+rX
    export ALTERNATE_OWNER\=sean
    export ALWAYS_EMBED_SWIFT_STANDARD_LIBRARIES\=NO
    export ALWAYS_SEARCH_USER_PATHS\=YES
    export APPLE_INTERNAL_DEVELOPER_DIR\=/AppleInternal/Developer
    export APPLE_INTERNAL_DIR\=/AppleInternal
    export APPLE_INTERNAL_DOCUMENTATION_DIR\=/AppleInternal/Documentation
    export APPLE_INTERNAL_LIBRARY_DIR\=/AppleInternal/Library
    export APPLE_INTERNAL_TOOLS\=/AppleInternal/Developer/Tools
    export APPLICATION_EXTENSION_API_ONLY\=NO
    export APPLY_RULES_IN_COPY_FILES\=NO
    export APPLY_RULES_IN_COPY_HEADERS\=NO
    export APP_SHORTCUTS_ENABLE_FLEXIBLE_MATCHING\=YES
    export ARCHS\=arm64
    export ARCHS_STANDARD\=arm64\ x86_64
    export ARCHS_STANDARD_32_64_BIT\=arm64\ x86_64\ i386
    export ARCHS_STANDARD_32_BIT\=i386
    export ARCHS_STANDARD_64_BIT\=arm64\ x86_64
    export ARCHS_STANDARD_INCLUDING_64_BIT\=arm64\ x86_64
    export ASSETCATALOG_COMPILER_GENERATE_ASSET_SYMBOLS\=YES
    export AUTOMATICALLY_MERGE_DEPENDENCIES\=NO
    export AVAILABLE_PLATFORMS\=appletvos\ appletvsimulator\ driverkit\ iphoneos\ iphonesimulator\ macosx\ watchos\ watchsimulator\ xros\ xrsimulator
    export BITCODE_GENERATION_MODE\=marker
    export BUILD_ACTIVE_RESOURCES_ONLY\=NO
    export BUILD_COMPONENTS\=headers\ build
    export BUILD_DIR\=/Volumes/Hub_SSD/Unreal\ Projects\ 2024/PCG_EX_C/Intermediate/ProjectFiles/build
    export BUILD_LIBRARY_FOR_DISTRIBUTION\=NO
    export BUILD_ROOT\=/Volumes/Hub_SSD/Unreal\ Projects\ 2024/PCG_EX_C/Intermediate/ProjectFiles/build
    export BUILD_STYLE\=
    export BUILD_VARIANTS\=normal
    export BUILT_PRODUCTS_DIR\=/Volumes/Hub_SSD/Unreal\ Projects\ 2024/PCG_EX_C/Intermediate/ProjectFiles/build/Development
    export BUNDLE_CONTENTS_FOLDER_PATH\=Contents/
    export BUNDLE_CONTENTS_FOLDER_PATH_deep\=Contents/
    export BUNDLE_EXECUTABLE_FOLDER_NAME_deep\=MacOS
    export BUNDLE_EXECUTABLE_FOLDER_PATH\=Contents/MacOS
    export BUNDLE_EXTENSIONS_FOLDER_PATH\=Contents/Extensions
    export BUNDLE_FORMAT\=deep
    export BUNDLE_FRAMEWORKS_FOLDER_PATH\=Contents/Frameworks
    export BUNDLE_PLUGINS_FOLDER_PATH\=Contents/PlugIns
    export BUNDLE_PRIVATE_HEADERS_FOLDER_PATH\=Contents/PrivateHeaders
    export BUNDLE_PUBLIC_HEADERS_FOLDER_PATH\=Contents/Headers
    export CACHE_ROOT\=/var/folders/hb/nh8pmp_53z1fkf2y115pmz8w0000gn/C/com.apple.DeveloperTools/15.2-15C5500c/Xcode
    export CLANG_COMPILE_CACHE_PATH\=/Users/sean/Library/Developer/Xcode/DerivedData/CompilationCache.noindex
    export CLANG_ENABLE_EXPLICIT_MODULES\=NO
    export CLANG_MODULES_BUILD_SESSION_FILE\=/Users/sean/Library/Developer/Xcode/DerivedData/ModuleCache.noindex/Session.modulevalidation
    export CLASS_FILE_DIR\=/Volumes/Hub_SSD/Unreal\ Projects\ 2024/PCG_EX_C/Intermediate/ProjectFiles/build/PCG_EX_C\ \(Mac\).build/Development/PCG_EX_C_Build.build/JavaClasses
    export CLONE_HEADERS\=NO
    export CODE_SIGNING_ALLOWED\=NO
    export CODE_SIGN_IDENTITY\=-
    export CODE_SIGN_INJECT_BASE_ENTITLEMENTS\=YES
    export COLOR_DIAGNOSTICS\=NO
    export COMPILER_INDEX_STORE_ENABLE\=Default
    export COMPRESS_PNG_FILES\=NO
    export CONFIGURATION\=Development
    export CONFIGURATION_BUILD_DIR\=/Volumes/Hub_SSD/Unreal\ Projects\ 2024/PCG_EX_C/Intermediate/ProjectFiles/build/Development
    export CONFIGURATION_TEMP_DIR\=/Volumes/Hub_SSD/Unreal\ Projects\ 2024/PCG_EX_C/Intermediate/ProjectFiles/build/PCG_EX_C\ \(Mac\).build/Development
    export COPYING_PRESERVES_HFS_DATA\=NO
    export COPY_HEADERS_RUN_UNIFDEF\=NO
    export COPY_PHASE_STRIP\=YES
    export CREATE_INFOPLIST_SECTION_IN_BINARY\=NO
    export DEAD_CODE_STRIPPING\=NO
    export DEBUGGING_SYMBOLS\=YES
    export DEFAULT_COMPILER\=com.apple.compilers.llvm.clang.1_0
    export DEFAULT_DEXT_INSTALL_PATH\=/System/Library/DriverExtensions
    export DEFAULT_KEXT_INSTALL_PATH\=/System/Library/Extensions
    export DEFINES_MODULE\=NO
    export DEPLOYMENT_LOCATION\=NO
    export DEPLOYMENT_POSTPROCESSING\=NO
    export DEPLOYMENT_TARGET_SETTING_NAME\=MACOSX_DEPLOYMENT_TARGET
    export DERIVED_FILES_DIR\=
    export DERIVED_FILE_DIR\=/Volumes/Hub_SSD/Unreal\ Projects\ 2024/PCG_EX_C/Intermediate/ProjectFiles/build/PCG_EX_C\ \(Mac\).build/Development/PCG_EX_C_Build.build/DerivedSources
    export DERIVED_SOURCES_DIR\=/Volumes/Hub_SSD/Unreal\ Projects\ 2024/PCG_EX_C/Intermediate/ProjectFiles/build/PCG_EX_C\ \(Mac\).build/Development/PCG_EX_C_Build.build/DerivedSources
    export DEVELOPER_APPLICATIONS_DIR\=/Applications/Xcode-beta.app/Contents/Developer/Applications
    export DEVELOPER_BIN_DIR\=/Applications/Xcode-beta.app/Contents/Developer/usr/bin
    export DEVELOPER_DIR\=/Applications/Xcode-beta.app/Contents/Developer
    export DEVELOPER_FRAMEWORKS_DIR\=/Applications/Xcode-beta.app/Contents/Developer/Library/Frameworks
    export DEVELOPER_FRAMEWORKS_DIR_QUOTED\=/Applications/Xcode-beta.app/Contents/Developer/Library/Frameworks
    export DEVELOPER_LIBRARY_DIR\=/Applications/Xcode-beta.app/Contents/Developer/Library
    export DEVELOPER_SDK_DIR\=/Applications/Xcode-beta.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs
    export DEVELOPER_TOOLS_DIR\=/Applications/Xcode-beta.app/Contents/Developer/Tools
    export DEVELOPER_USR_DIR\=/Applications/Xcode-beta.app/Contents/Developer/usr
    export DEVELOPMENT_LANGUAGE\=English
    export DONT_GENERATE_INFOPLIST_FILE\=NO
    export DO_HEADER_SCANNING_IN_JAM\=NO
    export DSTROOT\=/tmp/PCG_EX_C\ \(Mac\).dst
    export DT_TOOLCHAIN_DIR\=/Applications/Xcode-beta.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain
    export DWARF_DSYM_FILE_NAME\=.dSYM
    export DWARF_DSYM_FILE_SHOULD_ACCOMPANY_PRODUCT\=NO
    export DWARF_DSYM_FOLDER_PATH\=/Volumes/Hub_SSD/Unreal\ Projects\ 2024/PCG_EX_C/Intermediate/ProjectFiles/build/Development
    export EAGER_LINKING\=NO
    export EMBEDDED_CONTENT_CONTAINS_SWIFT\=NO
    export EMBEDDED_PROFILE_NAME\=embedded.provisionprofile
    export EMBED_ASSET_PACKS_IN_PRODUCT_BUNDLE\=NO
    export ENABLE_APP_SANDBOX\=NO
    export ENABLE_BITCODE\=NO
    export ENABLE_CODE_COVERAGE\=YES
    export ENABLE_DEFAULT_HEADER_SEARCH_PATHS\=YES
    export ENABLE_DEFAULT_SEARCH_PATHS\=YES
    export ENABLE_HARDENED_RUNTIME\=NO
    export ENABLE_HEADER_DEPENDENCIES\=YES
    export ENABLE_ON_DEMAND_RESOURCES\=NO
    export ENABLE_PREVIEWS\=NO
    export ENABLE_TESTABILITY\=NO
    export ENABLE_TESTING_SEARCH_PATHS\=NO
    export ENABLE_USER_SCRIPT_SANDBOXING\=NO
    export ENABLE_XOJIT_PREVIEWS\=NO
    export ENTITLEMENTS_DESTINATION\=Signature
    export EXCLUDED_INSTALLSRC_SUBDIRECTORY_PATTERNS\=.DS_Store\ .svn\ .git\ .hg\ CVS
    export EXCLUDED_RECURSIVE_SEARCH_PATH_SUBDIRECTORIES\=\*.nib\ \*.lproj\ \*.framework\ \*.gch\ \*.xcode\*\ \*.xcassets\ \(\*\)\ .DS_Store\ CVS\ .svn\ .git\ .hg\ \*.pbproj\ \*.pbxproj
    export FILE_LIST\=/Volumes/Hub_SSD/Unreal\ Projects\ 2024/PCG_EX_C/Intermediate/ProjectFiles/build/PCG_EX_C\ \(Mac\).build/Development/PCG_EX_C_Build.build/Objects/LinkFileList
    export FRAMEWORK_VERSION\=A
    export FUSE_BUILD_PHASES\=YES
    export FUSE_BUILD_SCRIPT_PHASES\=NO
    export GCC3_VERSION\=3.3
    export GCC_VERSION\=com.apple.compilers.llvm.clang.1_0
    export GCC_WARN_64_TO_32_BIT_CONVERSION\=NO
    export GENERATED_MODULEMAP_DIR\=/Volumes/Hub_SSD/Unreal\ Projects\ 2024/PCG_EX_C/Intermediate/ProjectFiles/build/GeneratedModuleMaps
    export GENERATE_INFOPLIST_FILE\=NO
    export GENERATE_INTERMEDIATE_TEXT_BASED_STUBS\=YES
    export GENERATE_MASTER_OBJECT_FILE\=NO
    export GENERATE_PKGINFO_FILE\=NO
    export GENERATE_PROFILING_CODE\=NO
    export GENERATE_TEXT_BASED_STUBS\=NO
    export GID\=20
    export GROUP\=staff
    export HEADERMAP_INCLUDES_FLAT_ENTRIES_FOR_TARGET_BEING_BUILT\=YES
    export HEADERMAP_INCLUDES_FRAMEWORK_ENTRIES_FOR_ALL_PRODUCT_TYPES\=YES
    export HEADERMAP_INCLUDES_NONPUBLIC_NONPRIVATE_HEADERS\=YES
    export HEADERMAP_INCLUDES_PROJECT_HEADERS\=YES
    export HEADERMAP_USES_FRAMEWORK_PREFIX_ENTRIES\=YES
    export HEADERMAP_USES_VFS\=NO
    export HIDE_BITCODE_SYMBOLS\=YES
    export HOME\=/Users/sean
    export HOST_ARCH\=arm64
    export HOST_PLATFORM\=macosx
    export INFOPLIST_ENABLE_CFBUNDLEICONS_MERGE\=YES
    export INFOPLIST_EXPAND_BUILD_SETTINGS\=YES
    export INFOPLIST_OUTPUT_FORMAT\=same-as-input
    export INFOPLIST_PREPROCESS\=NO
    export INLINE_PRIVATE_FRAMEWORKS\=NO
    export INSTALLHDRS_COPY_PHASE\=NO
    export INSTALLHDRS_SCRIPT_PHASE\=NO
    export INSTALL_DIR\=/tmp/PCG_EX_C\ \(Mac\).dst
    export INSTALL_GROUP\=staff
    export INSTALL_MODE_FLAG\=u+w,go-w,a+rX
    export INSTALL_OWNER\=sean
    export INSTALL_ROOT\=/tmp/PCG_EX_C\ \(Mac\).dst
    export JAVA_APP_STUB\=/System/Library/Frameworks/JavaVM.framework/Resources/MacOS/JavaApplicationStub
    export JAVA_ARCHIVE_CLASSES\=YES
    export JAVA_ARCHIVE_TYPE\=JAR
    export JAVA_COMPILER\=/usr/bin/javac
    export JAVA_FRAMEWORK_RESOURCES_DIRS\=Resources
    export JAVA_JAR_FLAGS\=cv
    export JAVA_SOURCE_SUBDIR\=.
    export JAVA_USE_DEPENDENCIES\=YES
    export JAVA_ZIP_FLAGS\=-urg
    export KEEP_PRIVATE_EXTERNS\=NO
    export LD_DEPENDENCY_INFO_FILE\=//_dependency_info.dat
    export LD_EXPORT_SYMBOLS\=YES
    export LD_GENERATE_MAP_FILE\=NO
    export LD_MAP_FILE_PATH\=/Volumes/Hub_SSD/Unreal\ Projects\ 2024/PCG_EX_C/Intermediate/ProjectFiles/build/PCG_EX_C\ \(Mac\).build/Development/PCG_EX_C_Build.build/-LinkMap--.txt
    export LD_NO_PIE\=NO
    export LD_QUOTE_LINKER_ARGUMENTS_FOR_COMPILER_DRIVER\=YES
    export LEGACY_DEVELOPER_DIR\=/Applications/Xcode-beta.app/Contents/PlugIns/Xcode3Core.ideplugin/Contents/SharedSupport/Developer
    export LIBRARY_DEXT_INSTALL_PATH\=/Library/DriverExtensions
    export LIBRARY_FLAG_NOSPACE\=YES
    export LIBRARY_KEXT_INSTALL_PATH\=/Library/Extensions
    export LINKER_DISPLAYS_MANGLED_NAMES\=NO
    export LINK_FILE_LIST_normal_arm64\=/Volumes/Hub_SSD/Unreal\ Projects\ 2024/PCG_EX_C/Intermediate/ProjectFiles/build/PCG_EX_C\ \(Mac\).build/Development/PCG_EX_C_Build.build/Objects-normal/arm64/.LinkFileList
    export LINK_WITH_STANDARD_LIBRARIES\=YES
    export LLVM_TARGET_TRIPLE_OS_VERSION\=macos14.2
    export LLVM_TARGET_TRIPLE_VENDOR\=apple
    export LOCALIZATION_EXPORT_SUPPORTED\=YES
    export LOCALIZATION_PREFERS_STRING_CATALOGS\=NO
    export LOCALIZED_STRING_MACRO_NAMES\=NSLocalizedString\ CFCopyLocalizedString
    export LOCALIZED_STRING_SWIFTUI_SUPPORT\=YES
    export LOCAL_ADMIN_APPS_DIR\=/Applications/Utilities
    export LOCAL_APPS_DIR\=/Applications
    export LOCAL_DEVELOPER_DIR\=/Library/Developer
    export LOCAL_LIBRARY_DIR\=/Library
    export LOCROOT\=/Volumes/Hub_SSD/Unreal\ Projects\ 2024/PCG_EX_C/Intermediate/ProjectFiles
    export LOCSYMROOT\=/Volumes/Hub_SSD/Unreal\ Projects\ 2024/PCG_EX_C/Intermediate/ProjectFiles
    export MACOSX_DEPLOYMENT_TARGET\=14.2
    export MAC_OS_X_PRODUCT_BUILD_VERSION\=23E224
    export MAC_OS_X_VERSION_ACTUAL\=140401
    export MAC_OS_X_VERSION_MAJOR\=140000
    export MAC_OS_X_VERSION_MINOR\=140400
    export MAKE_MERGEABLE\=NO
    export MERGEABLE_LIBRARY\=NO
    export MERGED_BINARY_TYPE\=none
    export MERGE_LINKED_LIBRARIES\=NO
    export METAL_LIBRARY_FILE_BASE\=default
    export METAL_LIBRARY_OUTPUT_DIR\=/Volumes/Hub_SSD/Unreal\ Projects\ 2024/PCG_EX_C/Intermediate/ProjectFiles/build/Development/
    export MODULE_CACHE_DIR\=/Users/sean/Library/Developer/Xcode/DerivedData/ModuleCache.noindex
    export NATIVE_ARCH\=arm64
    export NATIVE_ARCH_32_BIT\=arm
    export NATIVE_ARCH_64_BIT\=arm64
    export NATIVE_ARCH_ACTUAL\=arm64
    export OBJECT_FILE_DIR\=/Volumes/Hub_SSD/Unreal\ Projects\ 2024/PCG_EX_C/Intermediate/ProjectFiles/build/PCG_EX_C\ \(Mac\).build/Development/PCG_EX_C_Build.build/Objects
    export OBJROOT\=/Volumes/Hub_SSD/Unreal\ Projects\ 2024/PCG_EX_C/Intermediate/ProjectFiles/build
    export ONLY_ACTIVE_ARCH\=YES
    export OS\=MACOS
    export PATH\=/Applications/Xcode-beta.app/Contents/Developer/Library/Xcode/Plug-ins/XCBSpecifications.ideplugin/Contents/Resources:/Applications/Xcode-beta.app/Contents/Developer/Library/Xcode/Plug-ins/XCBSpecifications.ideplugin:/Applications/Xcode-beta.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin:/Applications/Xcode-beta.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/appleinternal/bin:/Applications/Xcode-beta.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/local/bin:/Applications/Xcode-beta.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/libexec:/Applications/Xcode-beta.app/Contents/Developer/Platforms/MacOSX.platform/usr/bin:/Applications/Xcode-beta.app/Contents/Developer/Platforms/MacOSX.platform/usr/appleinternal/bin:/Applications/Xcode-beta.app/Contents/Developer/Platforms/MacOSX.platform/usr/local/bin:/Applications/Xcode-beta.app/Contents/Developer/Platforms/MacOSX.platform/Developer/usr/bin:/Applications/Xcode-beta.app/Contents/Developer/Platforms/MacOSX.platform/Developer/usr/local/bin:/Applications/Xcode-beta.app/Contents/Developer/usr/bin:/Applications/Xcode-beta.app/Contents/Developer/usr/local/bin:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin
    export PATH_PREFIXES_EXCLUDED_FROM_HEADER_DEPENDENCIES\=/usr/include\ /usr/local/include\ /System/Library/Frameworks\ /System/Library/PrivateFrameworks\ /Applications/Xcode-beta.app/Contents/Developer/Headers\ /Applications/Xcode-beta.app/Contents/Developer/SDKs\ /Applications/Xcode-beta.app/Contents/Developer/Platforms
    export PER_ARCH_OBJECT_FILE_DIR\=/
    export PLATFORM_DEVELOPER_APPLICATIONS_DIR\=/Applications/Xcode-beta.app/Contents/Developer/Applications
    export PLATFORM_DEVELOPER_BIN_DIR\=/Applications/Xcode-beta.app/Contents/Developer/usr/bin
    export PLATFORM_DEVELOPER_LIBRARY_DIR\=/Applications/Xcode-beta.app/Contents/Developer/Library
    export PLATFORM_DEVELOPER_SDK_DIR\=/Applications/Xcode-beta.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs
    export PLATFORM_DEVELOPER_TOOLS_DIR\=/Applications/Xcode-beta.app/Contents/Developer/Tools
    export PLATFORM_DEVELOPER_USR_DIR\=/Applications/Xcode-beta.app/Contents/Developer/usr
    export PLATFORM_DIR\=/Applications/Xcode-beta.app/Contents/Developer/Platforms/MacOSX.platform
    export PLATFORM_DISPLAY_NAME\=macOS
    export PLATFORM_FAMILY_NAME\=macOS
    export PLATFORM_NAME\=macosx
    export PLATFORM_PREFERRED_ARCH\=x86_64
    export PLIST_FILE_OUTPUT_FORMAT\=same-as-input
    export PRECOMPS_INCLUDE_HEADERS_FROM_BUILT_PRODUCTS_DIR\=YES
    export PRECOMP_DESTINATION_DIR\=/Volumes/Hub_SSD/Unreal\ Projects\ 2024/PCG_EX_C/Intermediate/ProjectFiles/build/PCG_EX_C\ \(Mac\).build/Development/PCG_EX_C_Build.build/PrefixHeaders
    export PRESERVE_DEAD_CODE_INITS_AND_TERMS\=NO
    export PRODUCT_NAME\=
    export PROFILING_CODE\=NO
    export PROJECT\=PCG_EX_C\ \(Mac\)
    export PROJECT_DERIVED_FILE_DIR\=/Volumes/Hub_SSD/Unreal\ Projects\ 2024/PCG_EX_C/Intermediate/ProjectFiles/build/PCG_EX_C\ \(Mac\).build/DerivedSources
    export PROJECT_DIR\=/Volumes/Hub_SSD/Unreal\ Projects\ 2024/PCG_EX_C/Intermediate/ProjectFiles
    export PROJECT_FILE_PATH\=/Volumes/Hub_SSD/Unreal\ Projects\ 2024/PCG_EX_C/Intermediate/ProjectFiles/PCG_EX_C\ \(Mac\).xcodeproj
    export PROJECT_NAME\=PCG_EX_C\ \(Mac\)
    export PROJECT_TEMP_DIR\=/Volumes/Hub_SSD/Unreal\ Projects\ 2024/PCG_EX_C/Intermediate/ProjectFiles/build/PCG_EX_C\ \(Mac\).build
    export PROJECT_TEMP_ROOT\=/Volumes/Hub_SSD/Unreal\ Projects\ 2024/PCG_EX_C/Intermediate/ProjectFiles/build
    export REMOVE_CVS_FROM_RESOURCES\=YES
    export REMOVE_GIT_FROM_RESOURCES\=YES
    export REMOVE_HEADERS_FROM_EMBEDDED_BUNDLES\=YES
    export REMOVE_HG_FROM_RESOURCES\=YES
    export REMOVE_SVN_FROM_RESOURCES\=YES
    export SCAN_ALL_SOURCE_FILES_FOR_INCLUDES\=NO
    export SDKROOT\=/Applications/Xcode-beta.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX14.2.sdk
    export SDK_STAT_CACHE_ENABLE\=YES
    export SEPARATE_STRIP\=NO
    export SEPARATE_SYMBOL_EDIT\=NO
    export SHARED_DERIVED_FILE_DIR\=/Volumes/Hub_SSD/Unreal\ Projects\ 2024/PCG_EX_C/Intermediate/ProjectFiles/build/Development/DerivedSources
    export SHARED_PRECOMPS_DIR\=/Volumes/Hub_SSD/Unreal\ Projects\ 2024/PCG_EX_C/Intermediate/ProjectFiles/build/SharedPrecompiledHeaders
    export SKIP_INSTALL\=YES
    export SOURCE_ROOT\=/Volumes/Hub_SSD/Unreal\ Projects\ 2024/PCG_EX_C/Intermediate/ProjectFiles
    export SRCROOT\=/Volumes/Hub_SSD/Unreal\ Projects\ 2024/PCG_EX_C/Intermediate/ProjectFiles
    export STRINGSDATA_DIR\=/
    export STRINGSDATA_ROOT\=/Volumes/Hub_SSD/Unreal\ Projects\ 2024/PCG_EX_C/Intermediate/ProjectFiles/build/PCG_EX_C\ \(Mac\).build/Development/PCG_EX_C_Build.build
    export STRINGS_FILE_INFOPLIST_RENAME\=YES
    export STRINGS_FILE_OUTPUT_ENCODING\=UTF-16
    export STRIP_BITCODE_FROM_COPIED_FILES\=NO
    export STRIP_INSTALLED_PRODUCT\=YES
    export STRIP_PNG_TEXT\=NO
    export STRIP_STYLE\=all
    export STRIP_SWIFT_SYMBOLS\=YES
    export SUPPORTED_PLATFORMS\=macosx\ iphonesimulator\ iphoneos\ appletvsimulator\ appletvos
    export SUPPORTS_TEXT_BASED_API\=NO
    export SUPPRESS_WARNINGS\=NO
    export SWIFT_EMIT_LOC_STRINGS\=NO
    export SWIFT_PLATFORM_TARGET_PREFIX\=macos
    export SWIFT_RESPONSE_FILE_PATH_normal_arm64\=/Volumes/Hub_SSD/Unreal\ Projects\ 2024/PCG_EX_C/Intermediate/ProjectFiles/build/PCG_EX_C\ \(Mac\).build/Development/PCG_EX_C_Build.build/Objects-normal/arm64/.SwiftFileList
    export SYMROOT\=/Volumes/Hub_SSD/Unreal\ Projects\ 2024/PCG_EX_C/Intermediate/ProjectFiles/build
    export SYSTEM_ADMIN_APPS_DIR\=/Applications/Utilities
    export SYSTEM_APPS_DIR\=/Applications
    export SYSTEM_CORE_SERVICES_DIR\=/System/Library/CoreServices
    export SYSTEM_DEMOS_DIR\=/Applications/Extras
    export SYSTEM_DEVELOPER_APPS_DIR\=/Applications/Xcode-beta.app/Contents/Developer/Applications
    export SYSTEM_DEVELOPER_BIN_DIR\=/Applications/Xcode-beta.app/Contents/Developer/usr/bin
    export SYSTEM_DEVELOPER_DEMOS_DIR\=/Applications/Xcode-beta.app/Contents/Developer/Applications/Utilities/Built\ Examples
    export SYSTEM_DEVELOPER_DIR\=/Applications/Xcode-beta.app/Contents/Developer
    export SYSTEM_DEVELOPER_DOC_DIR\=/Applications/Xcode-beta.app/Contents/Developer/ADC\ Reference\ Library
    export SYSTEM_DEVELOPER_GRAPHICS_TOOLS_DIR\=/Applications/Xcode-beta.app/Contents/Developer/Applications/Graphics\ Tools
    export SYSTEM_DEVELOPER_JAVA_TOOLS_DIR\=/Applications/Xcode-beta.app/Contents/Developer/Applications/Java\ Tools
    export SYSTEM_DEVELOPER_PERFORMANCE_TOOLS_DIR\=/Applications/Xcode-beta.app/Contents/Developer/Applications/Performance\ Tools
    export SYSTEM_DEVELOPER_RELEASENOTES_DIR\=/Applications/Xcode-beta.app/Contents/Developer/ADC\ Reference\ Library/releasenotes
    export SYSTEM_DEVELOPER_TOOLS\=/Applications/Xcode-beta.app/Contents/Developer/Tools
    export SYSTEM_DEVELOPER_TOOLS_DOC_DIR\=/Applications/Xcode-beta.app/Contents/Developer/ADC\ Reference\ Library/documentation/DeveloperTools
    export SYSTEM_DEVELOPER_TOOLS_RELEASENOTES_DIR\=/Applications/Xcode-beta.app/Contents/Developer/ADC\ Reference\ Library/releasenotes/DeveloperTools
    export SYSTEM_DEVELOPER_USR_DIR\=/Applications/Xcode-beta.app/Contents/Developer/usr
    export SYSTEM_DEVELOPER_UTILITIES_DIR\=/Applications/Xcode-beta.app/Contents/Developer/Applications/Utilities
    export SYSTEM_DEXT_INSTALL_PATH\=/System/Library/DriverExtensions
    export SYSTEM_DOCUMENTATION_DIR\=/Library/Documentation
    export SYSTEM_KEXT_INSTALL_PATH\=/System/Library/Extensions
    export SYSTEM_LIBRARY_DIR\=/System/Library
    export TAPI_DEMANGLE\=YES
    export TAPI_ENABLE_PROJECT_HEADERS\=NO
    export TAPI_LANGUAGE\=objective-c
    export TAPI_LANGUAGE_STANDARD\=compiler-default
    export TAPI_VERIFY_MODE\=Pedantic
    export TARGETNAME\=PCG_EX_C_Build
    export TARGET_BUILD_DIR\=/Volumes/Hub_SSD/Unreal\ Projects\ 2024/PCG_EX_C/Intermediate/ProjectFiles/build/Development
    export TARGET_NAME\=PCG_EX_C_Build
    export TARGET_TEMP_DIR\=/Volumes/Hub_SSD/Unreal\ Projects\ 2024/PCG_EX_C/Intermediate/ProjectFiles/build/PCG_EX_C\ \(Mac\).build/Development/PCG_EX_C_Build.build
    export TEMP_DIR\=/Volumes/Hub_SSD/Unreal\ Projects\ 2024/PCG_EX_C/Intermediate/ProjectFiles/build/PCG_EX_C\ \(Mac\).build/Development/PCG_EX_C_Build.build
    export TEMP_FILES_DIR\=/Volumes/Hub_SSD/Unreal\ Projects\ 2024/PCG_EX_C/Intermediate/ProjectFiles/build/PCG_EX_C\ \(Mac\).build/Development/PCG_EX_C_Build.build
    export TEMP_FILE_DIR\=/Volumes/Hub_SSD/Unreal\ Projects\ 2024/PCG_EX_C/Intermediate/ProjectFiles/build/PCG_EX_C\ \(Mac\).build/Development/PCG_EX_C_Build.build
    export TEMP_ROOT\=/Volumes/Hub_SSD/Unreal\ Projects\ 2024/PCG_EX_C/Intermediate/ProjectFiles/build
    export TOOLCHAIN_DIR\=/Applications/Xcode-beta.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain
    export TREAT_MISSING_BASELINES_AS_TEST_FAILURES\=NO
    export UID\=501
    export UNINSTALLED_PRODUCTS_DIR\=/Volumes/Hub_SSD/Unreal\ Projects\ 2024/PCG_EX_C/Intermediate/ProjectFiles/build/UninstalledProducts
    export USER\=sean
    export USER_APPS_DIR\=/Users/sean/Applications
    export USER_LIBRARY_DIR\=/Users/sean/Library
    export USE_HEADERMAP\=YES
    export VALIDATE_DEVELOPMENT_ASSET_PATHS\=YES_ERROR
    export VALID_ARCHS\=arm64\ arm64e\ i386\ x86_64
    export WRAP_ASSET_PACKS_IN_SEPARATE_DIRECTORIES\=NO
    export XCODE_APP_SUPPORT_DIR\=/Applications/Xcode-beta.app/Contents/Developer/Library/Xcode
    export XCODE_PRODUCT_BUILD_VERSION\=15C5500c
    export XCODE_VERSION_ACTUAL\=1520
    export XCODE_VERSION_MAJOR\=1500
    export XCODE_VERSION_MINOR\=1520
    export arch\=undefined_arch
    export variant\=normal
    /Users/Shared/Epic\ Games/UE_5.3/Engine/Build/BatchFiles/Mac/XcodeBuild.sh PCG_EX_C macosx Development /Volumes/Hub_SSD/Unreal Projects 2024/PCG_EX_C/PCG_EX_C.uproject

Running Xcodebuild PCG_EX_C macosx Development /Volumes/Hub_SSD/Unreal Projects 2024/PCG_EX_C/PCG_EX_C.uproject
Setting up bundled DotNet SDK
/Users/Shared/Epic Games/UE_5.3/Engine/Build/BatchFiles/Mac/../../../Binaries/ThirdParty/DotNet/6.0.302/mac-arm64
Processing build for Target=PCG_EX_C Platform=Mac Configuration=Development -architecture=arm64 /Volumes/Hub_SSD/Unreal Projects 2024/PCG_EX_C/PCG_EX_C.uproject 
Running dotnet Engine/Binaries/DotNET/UnrealBuildTool/UnrealBuildTool.dll PCG_EX_C Mac Development /Volumes/Hub_SSD/Unreal Projects 2024/PCG_EX_C/PCG_EX_C.uproject -architecture=arm64
Log file: /Users/sean/Library/Application Support/Epic/UnrealBuildTool/Log.txt
Building PCG_EX_C...
Determining max actions to execute in parallel (10 physical cores, 10 logical cores)
  Executing up to 10 processes, one per physical core
------ Building 4 action(s) started ------
[1/4] Link [Apple] PCG_EX_C
Undefined symbols for architecture arm64:
  "typeinfo for FPCGContext", referenced from:
      typeinfo for FPCGExDebugContext in Module.PCGExtendedToolkit.7.cpp.o
      typeinfo for FPCGExPointsProcessorContext in Module.PCGExtendedToolkit.8.cpp.o
  "typeinfo for IPCGElement", referenced from:
      typeinfo for FPCGExCreateStateElement in Module.PCGExtendedToolkit.6.cpp.o
      typeinfo for FPCGExCreateNodeFilterElement in Module.PCGExtendedToolkit.7.cpp.o
      typeinfo for FPCGExCreateCustomGraphSocketElement in Module.PCGExtendedToolkit.7.cpp.o
      typeinfo for FPCGExGatherSocketsElement in Module.PCGExtendedToolkit.7.cpp.o
      typeinfo for FPCGExDotFilterDefinitionElement in Module.PCGExtendedToolkit.7.cpp.o
      typeinfo for FPCGExMeanFilterDefinitionElement in Module.PCGExtendedToolkit.7.cpp.o
      typeinfo for FPCGExNumericCompareFilterDefinitionElement in Module.PCGExtendedToolkit.7.cpp.o
      ...
  "typeinfo for UPCGSettings", referenced from:
      typeinfo for UPCGExCreateCustomGraphSocketSettings in Module.PCGExtendedToolkit.1.cpp.o
      typeinfo for UPCGExCreateStateSettings in Module.PCGExtendedToolkit.1.cpp.o
      typeinfo for UPCGExCreateNodeFilterSettings in Module.PCGExtendedToolkit.1.cpp.o
      typeinfo for UPCGExDotFilterDefinitionSettings in Module.PCGExtendedToolkit.2.cpp.o
      typeinfo for UPCGExDebugSettings in Module.PCGExtendedToolkit.2.cpp.o
      typeinfo for UPCGExForceGarbageCollectionSettings in Module.PCGExtendedToolkit.2.cpp.o
      typeinfo for UPCGExGatherSocketsSettings in Module.PCGExtendedToolkit.3.cpp.o
      ...
  "typeinfo for UPCGPointData", referenced from:
      typeinfo for UPCGExParamDataBase in Module.PCGExtendedToolkit.1.cpp.o
      typeinfo for UPCGExGraphDefinition in Module.PCGExtendedToolkit.3.cpp.o
  "typeinfo for FSimplePCGElement", referenced from:
      typeinfo for FPCGPointProcessingElementBase in Module.PCGExtendedToolkit.7.cpp.o
  "typeinfo for UObject", referenced from:
      typeinfo for UPCGExChamferCornerPoints in Module.PCGExtendedToolkit.1.cpp.o
      typeinfo for UPCGExOperation in Module.PCGExtendedToolkit.4.cpp.o
ld: symbol(s) not found for architecture arm64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
Total time in Parallel executor: 3.37 seconds
Total execution time: 4.09 seconds

Running Xcodebuild PCG_EX_C macosx Development /Volumes/Hub_SSD/Unreal Projects 2024/PCG_EX_C/PCG_EX_C.uproject

Setting up bundled DotNet SDK

/Users/Shared/Epic Games/UE_5.3/Engine/Build/BatchFiles/Mac/../../../Binaries/ThirdParty/DotNet/6.0.302/mac-arm64

Processing build for Target=PCG_EX_C Platform=Mac Configuration=Development -architecture=arm64 /Volumes/Hub_SSD/Unreal Projects 2024/PCG_EX_C/PCG_EX_C.uproject 

Running dotnet Engine/Binaries/DotNET/UnrealBuildTool/UnrealBuildTool.dll PCG_EX_C Mac Development /Volumes/Hub_SSD/Unreal Projects 2024/PCG_EX_C/PCG_EX_C.uproject -architecture=arm64

Log file: /Users/sean/Library/Application Support/Epic/UnrealBuildTool/Log.txt

Building PCG_EX_C...

Determining max actions to execute in parallel (10 physical cores, 10 logical cores)

  Executing up to 10 processes, one per physical core

------ Building 4 action(s) started ------

[1/4] Link [Apple] PCG_EX_C

Undefined symbols for architecture arm64:

  "typeinfo for FPCGContext", referenced from:

      typeinfo for FPCGExDebugContext in Module.PCGExtendedToolkit.7.cpp.o

      typeinfo for FPCGExPointsProcessorContext in Module.PCGExtendedToolkit.8.cpp.o

  "typeinfo for IPCGElement", referenced from:

      typeinfo for FPCGExCreateStateElement in Module.PCGExtendedToolkit.6.cpp.o

      typeinfo for FPCGExCreateNodeFilterElement in Module.PCGExtendedToolkit.7.cpp.o

      typeinfo for FPCGExCreateCustomGraphSocketElement in Module.PCGExtendedToolkit.7.cpp.o

      typeinfo for FPCGExGatherSocketsElement in Module.PCGExtendedToolkit.7.cpp.o

      typeinfo for FPCGExDotFilterDefinitionElement in Module.PCGExtendedToolkit.7.cpp.o

      typeinfo for FPCGExMeanFilterDefinitionElement in Module.PCGExtendedToolkit.7.cpp.o

      typeinfo for FPCGExNumericCompareFilterDefinitionElement in Module.PCGExtendedToolkit.7.cpp.o

      ...

  "typeinfo for UPCGSettings", referenced from:

      typeinfo for UPCGExCreateCustomGraphSocketSettings in Module.PCGExtendedToolkit.1.cpp.o

      typeinfo for UPCGExCreateStateSettings in Module.PCGExtendedToolkit.1.cpp.o

      typeinfo for UPCGExCreateNodeFilterSettings in Module.PCGExtendedToolkit.1.cpp.o

      typeinfo for UPCGExDotFilterDefinitionSettings in Module.PCGExtendedToolkit.2.cpp.o

      typeinfo for UPCGExDebugSettings in Module.PCGExtendedToolkit.2.cpp.o

      typeinfo for UPCGExForceGarbageCollectionSettings in Module.PCGExtendedToolkit.2.cpp.o

      typeinfo for UPCGExGatherSocketsSettings in Module.PCGExtendedToolkit.3.cpp.o

      ...

  "typeinfo for UPCGPointData", referenced from:

      typeinfo for UPCGExParamDataBase in Module.PCGExtendedToolkit.1.cpp.o

      typeinfo for UPCGExGraphDefinition in Module.PCGExtendedToolkit.3.cpp.o

  "typeinfo for FSimplePCGElement", referenced from:

      typeinfo for FPCGPointProcessingElementBase in Module.PCGExtendedToolkit.7.cpp.o

  "typeinfo for UObject", referenced from:

      typeinfo for UPCGExChamferCornerPoints in Module.PCGExtendedToolkit.1.cpp.o

      typeinfo for UPCGExOperation in Module.PCGExtendedToolkit.4.cpp.o

ld: symbol(s) not found for architecture arm64

Total time in Parallel executor: 3.37 seconds

Total execution time: 4.09 seconds

/Users/Shared/Epic Games/UE_5.3/clang:1:1: linker command failed with exit code 1 (use -v to see invocation)

**Comment by HatmanAlchemy** [comment]
Sorry if that is no help I can delete it

**Comment by HatmanAlchemy** [comment]
Hi TimYep tried this again and mistakenly sent you the old error screenshot which I then deleted and have now sent the new error on Github. I also sent the xcode error report ( a bit long sorry ) in case this has any clues. I can delete this off github if it takes too much spaceThanks for your helpSean On 16 Apr 2024, at 1:19 AM, Tim ***@***.***> wrote:﻿
@HatmanAlchemy any news on this?

**Comment by Nebukam** [comment]
Hey @HatmanAlchemy ! Sorry for the lag -- I've been looking at the error (the relevant part is basically things that come after `------ Building 4 action(s) started ------`).

I'm a bit puzzled because the first two issues are `FPCGContext` and `FPCGSettings` which are both vanilla PCG stuffs unrelated to PCGEx. Long story short it's as if the native PCG plugin doesn't exist within your project. What version of the engine do you use? Do you compile it yourself? 

I think it'd be worth bringing this up in the Unreal Source discord, there is a #pcg-framework channel with people that may be in a better position than me to answer why you get these errors.

**Comment by HatmanAlchemy** [comment]
Hi Tim. I made a new C++ project and enabled the 3 PCG plugins and all worked fine.
I put your plugin in a local to Project Plugin folder and it asks me to rebuild which I do in Xcode and it gives that error. I remove your plugin and it fails an Xcode build but builds fine through the launcher project

from Xcode I get this
dyld[58723]: Library not loaded: @rpath/libtbb.dylib
  Referenced from: <DE65A3B8-CCEF-38BD-86AC-61D6031C8E64> /Volumes/Hub_SSD/Unreal Projects 2024/PCG_Ex/Binaries/Mac/PCG_Ex.app/Contents/MacOS/PCG_Ex
  Reason: tried: '/Volumes/Hub_SSD/Unreal Projects 2024/PCG_Ex/Binaries/Mac/libtbb.dylib' (no such file), '/usr/lib/system/introspection/libtbb.dylib' (no such file, not in dyld cache), '/Volumes/Hub_SSD/Unreal Projects 2024/PCG_Ex/Binaries/Mac/PCG_Ex.app/Contents/MacOS/libtbb.dylib' (no such file), '/Volumes/Hub_SSD/Unreal Projects 2024/PCG_Ex/Binaries/Mac/PCG_Ex.app/Contents/MacOS/libtbb.dylib' (no such file), '/Volumes/Hub_SSD/Unreal Projects 2024/PCG_Ex/Binaries/Mac/libtbb.dylib' (no such file), '/Volumes/Hub_SSD/Unreal Projects 2024/PCG_Ex/Binaries/Mac/PCG_Ex.app/Contents/MacOS/../../../../../../Users/Shared/Epic Games/UE_5.3/Engine/Binaries/ThirdParty/Intel/TBB/Mac/libtbb.dylib' (no such file), '/Volumes/Hub_SSD/Unreal Projects 2024/PCG_Ex/Binaries/Mac/PCG_Ex.app/Contents/MacOS/../../Contents/UE/Engine/Binaries/ThirdParty/Intel/TBB/Mac/libtbb.dylib' (no such file), '/Volumes/Hub_SSD/Unreal Projects 2024/PCG_Ex/Binaries/Mac/PCG_Ex.app/Contents/MacOS/file://engine/Binaries/ThirdParty/Intel/TBB/Mac/libtbb.dylib' (no such file), '/Volumes/Hub_SSD/Unreal Projects 2024/PCG_Ex/Binaries/Mac/PCG_Ex.app/Contents/MacOS/../../../../../../Users/Shared/Epic Games/UE_5.3/Engine/Binaries/ThirdParty/Ogg/Mac/libtbb.dylib' (no such file), '/Volumes/Hub_SSD/Unreal Projects 2024/PCG_Ex/Binaries/Mac/PCG_Ex.app/Contents/MacOS/../../Contents/UE/Engine/Binaries/ThirdParty/Ogg/Mac/libtbb.dylib' (no such file), '/Volumes/Hub_SSD/Unreal Projects 2024/PCG_Ex/Binaries/Mac/PCG_Ex.app/Contents/MacOS/file://engine/Binaries/ThirdParty/Ogg/Mac/libtbb.dylib' (no such file), '/Volumes/Hub_SSD/Unreal Projects 2024/PCG_Ex/Binaries/Mac/PCG_Ex.app/Contents/MacOS/../../../../../../Users/Shared/Epic Games/UE_5.3/Engine/Binaries/ThirdParty/Vorbis/Mac/libtbb.dylib' (no such file), '/Volumes/Hub_SSD/Unreal Projects 2024/PCG_Ex/Binaries/Mac/PCG_Ex.app/Contents/MacOS/../../Contents/UE/Engine/Binaries/ThirdParty/Vorbis/Mac/libtbb.dylib' (no such file), '/Volumes/Hub_SSD/Unreal Projects 2024/PCG_Ex/Binaries/Mac/PCG_Ex.app/Contents/MacOS/file://engine/Binaries/ThirdParty/Vorbis/Mac/libtbb.dylib' (no such file), '/Volumes/Hub_SSD/Unreal Projects 2024/PCG_Ex/Binaries/Mac/PCG_Ex.app/Contents/MacOS/libtbb.dylib' (no such file), '/Volumes/Hub_SSD/Unreal Projects 2024/PCG_Ex/Binaries/Mac/PCG_Ex.app/Contents/MacOS/libtbb.dylib' (no such file), '/Volumes/Hub_SSD/Unreal Projects 2024/PCG_Ex/Binaries/Mac/libtbb.dylib' (no such file), '/Volumes/Hub_SSD/Unreal Projects 2024/PCG_Ex/Binaries/Mac/PCG_Ex.app/Contents/MacOS/../../../../../../Users/Shared/Epic Games/UE_5.3/Engine/Binaries/ThirdParty/Intel/TBB/Mac/libtbb.dylib' (no such file), '/Volumes/Hub_SSD/Unreal Projects 2024/PCG_Ex/Binaries/Mac/PCG_Ex.app/Contents/MacOS/../../Contents/UE/Engine/Binaries/ThirdParty/Intel/TBB/Mac/libtbb.dylib' (no such file), '/Volumes/Hub_SSD/Unreal Projects 2024/PCG_Ex/Binaries/Mac/PCG_Ex.app/Contents/MacOS/file://engine/Binaries/ThirdParty/Intel/TBB/Mac/libtbb.dylib' (no such file), '/Volumes/Hub_SSD/Unreal Projects 2024/PCG_Ex/Binaries/Mac/PCG_Ex.app/Contents/MacOS/../../../../../../Users/Shared/Epic Games/UE_5.3/Engine/Binaries/ThirdParty/Ogg/Mac/libtbb.dylib' (no such file), '/Volumes/Hub_SSD/Unreal Projects 2024/PCG_Ex/Binaries/Mac/PCG_Ex.app/Contents/MacOS/../../Contents/UE/Engine/Binaries/ThirdParty/Ogg/Mac/libtbb.dylib' (no such file), '/Volumes/Hub_SSD/Unreal Projects 2024/PCG_Ex/Binaries/Mac/PCG_Ex.app/Contents/MacOS/file://engine/Binaries/ThirdParty/Ogg/Mac/libtbb.dylib' (no such file), '/Volumes/Hub_SSD/Unreal Projects 2024/PCG_Ex/Binaries/Mac/PCG_Ex.app/Contents/MacOS/../../../../../../Users/Shared/Epic Games/UE_5.3/Engine/Binaries/ThirdParty/Vorbis/Mac/libtbb.dylib' (no such file), '/Volumes/Hub_SSD/Unreal Projects 2024/PCG_Ex/Binaries/Mac/PCG_Ex.app/Contents/MacOS/../../Contents/UE/Engine/Binaries/ThirdParty/Vorbis/Mac/libtbb.dylib' (no such file), '/Volumes/Hub_SSD/Unreal Projects 2024/PCG_Ex/Binaries/Mac/PCG_Ex.app/Contents/MacOS/file://engine/Binaries/ThirdParty/Vorbis/Mac/libtbb.dylib' (no such file)

**Comment by Nebukam** [comment]
Hmmm again none of these errors relate to PCGEx, I'm honestly clueless as to why you get these errors :/
Out of curiosity, does this happen with other CPP plugins as well? For example if you try including [SUDS](https://github.com/sinbad/SUDS) then remove it, do you end up with the same oddly unrelated errors?

**Comment by HatmanAlchemy** [comment]
OK it's probably a Mac and UE 5.3. Yes I have other code plugins that work. Thanks for your time and help. If I make a C++ code project it fails but launcher it runs. I don't know enough about Xcode to figure it out. Sean

**Comment by Nebukam** [comment]
I'm sorry I couldn't be more helpful >.<

**Comment by HatmanAlchemy** [comment]
Yes SUDS built and compiled fine 
![Screenshot 2024-04-19 at 4 26 35 PM](https://github.com/Nebukam/PCGExtendedToolkit/assets/21061155/207d51dc-7408-4c1b-a398-ca888a53cc4e)

**Comment by Nebukam** [comment]
Any chance you’re running 5.3 and not 5.3.2 ?

Le ven. 19 avr. 2024 à 11:28, HatmanAlchemy ***@***.***> a
écrit :

**Comment by HatmanAlchemy** [comment]
5.3.2 On 19 Apr 2024, at 5:26 PM, Tim ***@***.***> wrote:﻿
Any chance you’re running 5.3 and not 5.3.2 ?



Le ven. 19 avr. 2024 à 11:28, HatmanAlchemy ***@***.***> a

écrit :

**Comment by Nebukam** [comment]
I've tried to understand what's going on but I'm clueless at the moment; I'd need Apple hardware to do trial and error which sadly I don't have at the moment -- sadly this will be a "won't fix" for the short/mid term.

**Comment by HatmanAlchemy** [comment]
Hi Tim. It builds a lot further along now in 5.4 main and I get the following failure points

![Screenshot 2024-04-24 at 6 46 00 AM](https://github.com/Nebukam/PCGExtendedToolkit/assets/21061155/c62032f9-2fbb-483e-a00b-6a4b98fecfae)

**Comment by HatmanAlchemy** [comment]
Hi Tim. It builds a lot further on 5.4. Have left a message on githubOn 19 Apr 2024, at 6:08 PM, (null) (null) ***@***.***> wrote:﻿5.3.2 On 19 Apr 2024, at 5:26 PM, Tim ***@***.***> wrote:﻿
Any chance you’re running 5.3 and not 5.3.2 ?



Le ven. 19 avr. 2024 à 11:28, HatmanAlchemy ***@***.***> a

écrit :

**Comment by HatmanAlchemy** [comment]
Hi Tim - sent an update

**Comment by HatmanAlchemy** [comment]
Hi tim I retried building in 5.4 and it builds a lot further so the issue must have been 5.3. I sent an update to github but not sure if you getting these. On 25 Apr 2024, at 6:36 AM, (null) (null) ***@***.***> wrote:﻿Hi Tim. It builds a lot further on 5.4. Have left a message on githubOn 19 Apr 2024, at 6:08 PM, (null) (null) ***@***.***> wrote:﻿5.3.2 On 19 Apr 2024, at 5:26 PM, Tim ***@***.***> wrote:﻿
Any chance you’re running 5.3 and not 5.3.2 ?



Le ven. 19 avr. 2024 à 11:28, HatmanAlchemy ***@***.***> a

écrit :

**Comment by Nebukam** [comment]
Hey @HatmanAlchemy! I'm just finally getting  5.4 vanilla and looking more into this, but don't get your hopes up as it really sounds like this only happens on macOS :(

The only relevant part for me is whatever involves PCGEx files, so the bottom ones:
![image](https://github.com/Nebukam/PCGExtendedToolkit/assets/5935407/8f4055fd-75f1-48e8-b10f-a815b1aa5760)

**Comment by Nebukam** [comment]
Pushed an update!

**Comment by HatmanAlchemy** [comment]
Tried the update on a fresh 5.4 project but no luck. Thanks anyway. Would you like me to close the issue ?

Sean

**Comment by Nebukam** [comment]
Damn this is f’d up; have you removed the PCGEx plugin entierely or just
pulled the repo on top of the previous one? If so, try removing the
generated folders (intermediate + binaries). Unless the error is a new one,
it should be fixed since it’s « just » a missing import issue.

As for closing the issue, it’s still an actual valid issue and I’m really
hoping I’ll be able to find what’s wrong at some point so leave it open —
some people may be able to contribute more meaningfully than I have!

Le dim. 28 avr. 2024 à 12:23, HatmanAlchemy ***@***.***> a
écrit :

> Tried the update on a fresh 5.4 project but no luck. Thanks anyway. Would
> you like me to close the issue ?
>
> Sean
>

**Comment by HatmanAlchemy** [comment]
Yes tried a brand new 5.4 C++ project and a fresh unzipped plugin from your update. It builds a few GB of stuff ( more than it did in 5.3 )I have even tried it with Jetbrains Rider free trial copy to see if that would give more help/ clues but nothing I can make sense ofSeanOn 28 Apr 2024, at 5:28 PM, Tim ***@***.***> wrote:﻿
Damn this is f’d up; have you removed the PCGEx plugin entierely or just

pulled the repo on top of the previous one? If so, try removing the

generated folders (intermediate + binaries). Unless the error is a new one,

it should be fixed since it’s « just » a missing import issue.



As for closing the issue, it’s still an actual valid issue and I’m really

hoping I’ll be able to find what’s wrong at some point so leave it open —

some people may be able to contribute more meaningfully than I have!



Le dim. 28 avr. 2024 à 12:23, HatmanAlchemy ***@***.***> a

écrit :



> Tried the update on a fresh 5.4 project but no luck. Thanks anyway. Would

> you like me to close the issue ?

>

> Sean

>

**Comment by HatmanAlchemy** [comment]
Is it anything you can raise with the PCG team on my behalf ? There might be a mac user ( longshot ) It might be a PCG issue ?On 28 Apr 2024, at 8:42 PM, (null) (null) ***@***.***> wrote:﻿Yes tried a brand new 5.4 C++ project and a fresh unzipped plugin from your update. It builds a few GB of stuff ( more than it did in 5.3 )I have even tried it with Jetbrains Rider free trial copy to see if that would give more help/ clues but nothing I can make sense ofSeanOn 28 Apr 2024, at 5:28 PM, Tim ***@***.***> wrote:﻿
Damn this is f’d up; have you removed the PCGEx plugin entierely or just

pulled the repo on top of the previous one? If so, try removing the

generated folders (intermediate + binaries). Unless the error is a new one,

it should be fixed since it’s « just » a missing import issue.



As for closing the issue, it’s still an actual valid issue and I’m really

hoping I’ll be able to find what’s wrong at some point so leave it open —

some people may be able to contribute more meaningfully than I have!



Le dim. 28 avr. 2024 à 12:23, HatmanAlchemy ***@***.***> a

écrit :



> Tried the update on a fresh 5.4 project but no luck. Thanks anyway. Would

> you like me to close the issue ?

>

> Sean

>

**Comment by Nebukam** [comment]
@HatmanAlchemy I realized I missed one error that showed in XCode, if you get the latest do you still see these 3 errors?
![image](https://github.com/Nebukam/PCGExtendedToolkit/assets/5935407/d934ab85-a06f-4188-9ed0-c4ef65563d14)

If it's still failing but with new warnings it means we're progressing! :D

**Comment by HatmanAlchemy** [comment]
I get the first one 

![Screenshot 2024-04-28 at 9 00 10 PM](https://github.com/Nebukam/PCGExtendedToolkit/assets/21061155/6066ecf0-bf39-49d9-930d-c8d3becaedef)

**Comment by Nebukam** [comment]
Yeah this one I just submitted a fix for a few minutes ago it should be gone 👀

**Comment by HatmanAlchemy** [comment]
Sorry I get this now Which I can sometimes fix with a system reboot - it sometimes is a stray process running in the background  Rebooting again...
![Screenshot 2024-04-28 at 9 24 18 PM](https://github.com/Nebukam/PCGExtendedToolkit/assets/21061155/dc57f66a-ff4e-4274-b890-fe43b1d93536)

**Comment by HatmanAlchemy** [comment]
nope still get that clang - will try again tomorrow

**Comment by HatmanAlchemy** [comment]
Hi Tim - I hope you don't mind I have asked a friend Maxime Depart whose Shader World Asset I use a lot. He is a compatriot of yours and an excellent coder. He has a Mac and has kindly agreed to have a look at this for me as he can test it on his Mac and maybe find the issue for us Mac users.

**Comment by Nebukam** [comment]
Hi Sean! I would be more than happy to have someone look at this; please ask your friend to do a PR on the repo if you end up solving the problem ❤️

**Comment by HatmanAlchemy** [comment]
Maximes fork now builds and works on my Mac 
![Uploading Screenshot 2024-05-04 at 10.39.07 AM.png…]()

**Comment by Nebukam** [comment]
This is great to hear and see! I just merged the PR :)
Thank a lot to you both! ❤️

**Comment by HatmanAlchemy** [comment]
Hi Tim

Finally got around to using the PCG Extended Toolkit ( on my Mac )

I needed it to path trace from a start to end point and use the points to spawn Brushes on a Shader World ( Maxime Dupart marketplace asset ).

The brushes deform the landscape so I can build rivers. I can also use it to build roads.

All at runtime

I thought you might like to see.

Like this




Thanks for a great tool

Sean

**Comment by HatmanAlchemy** [comment]
![Uploading Screenshot 2024-06-16 at 11.59.45 AM.png…]()

**Comment by HatmanAlchemy** [comment]
![Screenshot 2024-06-16 at 11 59 45 AM](https://github.com/Nebukam/PCGExtendedToolkit/assets/21061155/e86b0ebd-9efa-43a8-9466-aa2939ac40b7)
